semester1/hw9/task3/9.3.cpp

```diff
@@ -0,0 +1,106 @@
+#include <cstring>
+#include <iostream>
+
+using namespace std;
+
+const int mod = (int)1e9 + 7;
+const int base = 101;
+
+int hash(char *s, int start, int length)
```

Comment:
/usr/include/c++/5/bits/functional_hash.h:58:12: note:                 template<class _Tp> struct std::hash
     struct hash;
            ^
9.3.cpp:73:19: error: reference to ‘hash’ is ambiguous
  int needleHash = hash(needle, 0, m);
                   ^
9.3.cpp:9:5: note: candidates are: int hash(char*, int, int)
 int hash(char *s, int start, int length)
     ^
In file included from /usr/include/c++/5/bits/basic_string.h:5471:0,
                 from /usr/include/c++/5/string:52,
                 from /usr/include/c++/5/bits/locale_classes.h:40,
                 from /usr/include/c++/5/bits/ios_base.h:41,
                 from /usr/include/c++/5/ios:42,
                 from /usr/include/c++/5/ostream:38,
                 from /usr/include/c++/5/iostream:39,
                 from 9.3.cpp:2:
/usr/include/c++/5/bits/functional_hash.h:58:12: note:                 template<class _Tp> struct std::hash
     struct hash;